Week04/MoviesApp/app/src/main/java/tech/danielwaiguru/moviesapp/viewmodels/UserViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package tech.danielwaiguru.moviesapp.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import tech.danielwaiguru.moviesapp.database.MovieDatabase
+import tech.danielwaiguru.moviesapp.models.User
+
+import tech.danielwaiguru.moviesapp.repositories.UserRepository
+
+class UserViewModel(application: Application): AndroidViewModel(application) {
+    private val userRepository: UserRepository
+    init {
+        val userDao = MovieDatabase.getDatabaseInstance(application, viewModelScope).userDao()
```

Comment:
pass in the repo to the class through the constructor, instead of using the init block